### PR TITLE
Add tk.turtlico.Turtlico

### PR DESCRIPTION
**Pull request by oksoft-git**

### CODE REVIEW

File: com.orsan.Turtlico.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        "--talk-name=org.freedesktop.Flatpak",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options": {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
+        "env": {
+            "V": "1"
+        }
+    },
```
Comment (bilelmoussaoui) [inline com.orsan.Turtlico.json]: Not needed anymore. Already set by default.

File: com.orsan.Turtlico.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        "/share/man",
+        "/share/pkgconfig",
+        "/share/vala",
+        "*.la",
+        "*.a"
+    ],
+    "modules": [
+        {
+            "name" : "gtksourceview",
+            "builddir": true,
```
Comment (bilelmoussaoui) [inline com.orsan.Turtlico.json]:
```suggestion
```

File: com.orsan.Turtlico.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            "buildsystem": "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/GNOME/gtksourceview.git"
```
Comment (bilelmoussaoui) [inline com.orsan.Turtlico.json]: Please use a tarball here.

File: com.orsan.Turtlico.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                }
+            ]
+	},
+        {
+            "name": "turtlico",
+            "builddir": true,
```
Comment (bilelmoussaoui) [inline com.orsan.Turtlico.json]:
```suggestion
```

File: com.orsan.Turtlico.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--filesystem=host",
+        "--talk-name=ca.desrt.dconf",
+        "--talk-name=org.freedesktop.Flatpak",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (bilelmoussaoui) [inline com.orsan.Turtlico.json]: Remove dconf permissions. They are not needed for 3.34.

File: com.orsan.Turtlico.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--filesystem=host",
+        "--talk-name=ca.desrt.dconf",
+        "--talk-name=org.freedesktop.Flatpak",
```
Comment (bilelmoussaoui) [inline com.orsan.Turtlico.json]: Why it needs to talk to Flatpak here?

**Comment by LeandroStanger** [comment]
bot, build com.orsan.Turtlico

**Comment by oksoft-git** [comment]
bot, build com.orsan.Turtlico

File: com.orsan.Turtlico.json (hunk 6 above)
Comment (oksoft-git) [reply com.orsan.Turtlico.json]: Because it will access GPIO from RPI etc

File: com.orsan.Turtlico.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    "runtime-version": "3.34",
+    "sdk": "org.gnome.Sdk",
+    "command": "turtlico",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
```
Comment (barthalion) [inline com.orsan.Turtlico.json]: These two dconf lines also should be removed. GLib >=2.60.6 stores settings locally, without host access.

**Comment by FakeShemp** [comment]
Is orsan.com down or does it not exist? Why not use https://turtlico.tk/

**Comment by LeandroStanger** [comment]
bot, build tk.turtlico.Turtlico

**Comment by bilelmoussaoui** [comment]
Are you the upstream maintainer ?

**Comment by saytamkenorh** [comment]
> Are you the upstream maintainer ?

Yes. I am the upstream maintainer.

File: tk.turtlico.Turtlico.json [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    "modules": [
+        {
+            "name" : "gtksourceview",
+            "builddir": true,
+            "buildsystem": "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/GNOME/gtksourceview.git",
+                    "branch" : "4.4.0"
```
Comment (bilelmoussaoui) [inline tk.turtlico.Turtlico.json]: Please specify a commit here too.

File: tk.turtlico.Turtlico.json [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        "/share/man",
+        "/share/pkgconfig",
+        "/share/vala",
+        "*.la",
+        "*.a"
+    ],
+    "modules": [
+        {
+            "name" : "gtksourceview",
+            "builddir": true,
```
Comment (bilelmoussaoui) [inline tk.turtlico.Turtlico.json]:
```suggestion
```

File: tk.turtlico.Turtlico.json [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+            "buildsystem": "meson",
+            "config-opts" : [
+                "-Dflatpak=true"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.com/turtlico/turtlico.git",
+		    "tag": "0.2.0",
+		    "commit": "0f8eeddbcd2b84d461255fd605ae4088f7cc99b9"
```
Comment (bilelmoussaoui) [inline tk.turtlico.Turtlico.json]: Nitpick: indentation is wrong here

**Comment by bilelmoussaoui** [comment]
bot, build tk.turtlico.Turtlico

**Comment by bilelmoussaoui** [comment]
The app runs fine,, except it asks for the permission to install missing dependencies? and runs a command with root to on the host. Is there any reason for doing this?
Screenshot from 2019-12-15 20-09-04

**Comment by saytamkenorh** [comment]
Turtlico is a tool for creating programs using icons (similar to Scratch). When you click the run button the icons are converted into a Python script. This script is then run outside of the sandbox. The main reason for doing this is unavailability of the gpiozero library. Also Python that is bundled in GNOME runtime does not contain Tk module. This could be solved by bundling custom build of Python, Tk and Tcl though.
So the app asks for installing packages that are required to run program created in it. That is Python 3 and its modules.

**Comment by bilelmoussaoui** [comment]
Yes please, include whatever needed to run the app in the flatpak package.

**Comment by LeandroStanger** [comment]
bot, build tk.turtlico.Turtlico

File: tk.turtlico.Turtlico.json [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+            "config-opts": [
+                "--with-x",
+                "--enable-threads"
+            ],
+            "post-install" : [
+                "chmod 755 /app/lib/libtk8.6.so"
+            ]
+        },
+        {
+            "name" : "python3",
```
Comment (bilelmoussaoui) [inline tk.turtlico.Turtlico.json]: Is python3 rebuilt here just for tkinter? if it's the case, you can just install it from pip sources without re-building  pyton3.

Comment (bilelmoussaoui) [reply tk.turtlico.Turtlico.json]: Never mind.

File: tk.turtlico.Turtlico.json [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+    "sdk": "org.gnome.Sdk",
+    "command": "turtlico",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=host",
+        "--device=all",
+        "--env=PYTHONPATH=/usr/lib/python3.7/site-packages",
```
Comment (TingPing) [inline tk.turtlico.Turtlico.json]: This shouldn't be needed, sounds like a bug to report to the runtime.

File: tk.turtlico.Turtlico.json [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+        "--env=GPIOZERO_PIN_FACTORY=native"
+    ],
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig",
+        "/lib/debug",
```
Comment (TingPing) [inline tk.turtlico.Turtlico.json]:
```suggestion
```

**Comment by saytamkenorh** [comment]
@TingPing @bilelmoussaoui Any update on this? The pull request is still labeled awaiting-changes but everything is marked as resolved.

**Comment by bilelmoussaoui** [comment]
bot, build tk.turtlico.Turtlico

**Pull request closed without merging**